### ISSUE TITLE
Iteration 1.

### ISSUE THREAD

**Issue by dominictarr** [dominictarr/feedopensource#4]
These are the features to be addressed in the first iteration,
as well as the prototype of the iteration tracker.

here are links to the tasks this iteration seeks to implement:
- get status of a task (open or closed)  
  https://github.com/dominictarr/feedopensource/issues/1
- generate a iteration status badge:  
  https://github.com/dominictarr/feedopensource/issues/2
- handle basic payments:
  dominictarr#3
- use https
  dominictarr/feedopensource#10

[![badge](https://feedopensource.com/iteration/dominictarr/feedopensource/1PTAwipYpP63uNrcxfm5FewxRdZyar6ceu.png#1.2)](https://feedopensource.com/iteration/dominictarr/feedopensource/1PTAwipYpP63uNrcxfm5FewxRdZyar6ceu#1.2)

(edit: added progress bar - note, this is being cached by github, because it's yet not served over https)
(edit: use all the link formats that github supports)
(edit: new badge)
I'm gonna be conservative and give this two weeks.
This is new stuff, so problems are likely to crop up,

I am only gonna ask for 1.2 btc, which very much for hiring a developer for that length of time, but the most important thing here is to validate the idea by testing whether people are actually prepared to pay, and 1.2 btc is enough to be a real test, but not too much either.

Of course, you'd probably never see a kickstarter for this little, but the whole point is that this is iterative, and so there is a real way to monitor the progress and provide meaningful feedback.

**Comment by ralphtheninja** [comment]
I'll back half of that. Making this repo public soon?

**Comment by dominictarr** [comment]
Awesome! yes I will make it public soon, just a few more small things to do.

This iteration is only the first step. We'll probably (guessing) need 3 or so to get to something that is easy to use on other projects, and maybe more than that to get something really usable, I don't really know yet because I havn't used the system yet.
So, i think it's best to fund an amount that you could afford to continue to fund for a while.
Like, just fund one 6th, but for 3 iterations.

**Comment by ralphtheninja** [comment]
Sounds good!

**Comment by NHQ** [comment]
johnny honestly backs this IT with a fifth of a bit

**Comment by dominictarr** [comment]
okay, I put in https://blockchain.info/tx/bc0fbffd92993aa3eed60bb69e7320283100ffea36c5d26f851a8ebcb616f592 just to check things where working before I go live...

**Comment by dominictarr** [comment]
_this is now live_

**Comment by ralphtheninja** [comment]
Yay!

**Comment by binarykitchen** [comment]
Awesome stuff! Curious to see if this is going to work ...

Just did a small transaction with the ID b669c10b631217487437b284f811dc97c9a7f45520ef8fc2a263d268bb3c8560-000

**Comment by ghost** [comment]
f064ade31e374a57ebd03d9435f30ca8c8809b068974317394f433d153828112

**Comment by ralphtheninja** [comment]
https://blockchain.info/tx/3010b4d93dda15df8345ede9164b9adf2e25e1ecb9d978c74e2cfdd36315c735

**Comment by bluntworks** [comment]
https://blockchain.info/tx/ac2afeb9400f401b67eb39f25d4b4d046d35308117f0db6dace7165c89af6c0d

**Comment by gotcha** [comment]
https://blockchain.info/tx/5cd0556cb8522fb3485a6e1f79f3e7091d2bdacc98c8fe9c8e097752548a6a6d

**Comment by julianduque** [comment]
https://blockchain.info/tx/eca34ac3d130d1fe29668c4bb9d10d7d08d1b22f62cae9488105303e1494d06e

**Comment by hansent** [comment]
GyX//+gwsjKPS/03enVZFEkmnavF794Dyfki/ATbHPEMRut6US/QhqqiX/D/yfUyHCJHUCIUWNbE2QgB8eDBIe4=

https://blockchain.info/tx/a92f09c0926f64479e060b24d0aa7f0af0f793ec81611f04e20e6c2a7f77c406

**Comment by levity** [comment]
https://blockchain.info/tx/3be0bfb93638f01007e023faf51e28a1c690c0cb0130d749d4c4dfbf6a8d339b

**Comment by dominictarr** [comment]
thanks, you guys rule!

**Comment by eugeneware** [comment]
http://blockchain.info/tx/f11a08dcc2e336095c5118bb95124be30e379aaee4a98a532e524e475fcf563b

**Comment by ralphtheninja** [comment]
Soon up to 1.2. Awesome!

**Comment by eins78** [comment]
:bell: _ding ding ding_ :bell: 
https://blockchain.info/tx/96ebb5159544d6c3a3dc89f6b6886812c23e012e849f6ad92cfa6aef2f491c8d

> Total Received    1.26125445 BTC

**Comment by ralphtheninja** [comment]
@eins78 Haha, nice amount too! :)

**Comment by dominictarr** [comment]
THANKS EVERYONE!

Okay, time to get to work!

I'll be posting progress here...

**Comment by dominictarr** [comment]
Okay, so this after noon/evening I've got code to parse iterations from the list of github issues,
and to parse contributions out of an iterations comments.

I spent a bit of time wrestling with the github api. I was briefly tempted to replicate github's data into a local leveldb - but, the way github is set up, that would be too hard. 
I abandoned that, and ended up doing it the simplest way. Incidentially, it's the same way as in
@ralphtheninja/btcprogress which we made when we originally started talking about this idea.

when generating a resource (png progress bar, etc) just fall through to the github/blockchain api - but with a microcache (say, 20 seconds) so that we don't go over rate limits.

This is super simple, and works completely fine.

Oh by the way, there is about 0.23 worth of payments unclaimed (you didn't post your transaction here) If you contributed but didn't claim please do so - there is nothing wrong with free money, I'd rather know who it is that gives a damn.

**Comment by dominictarr** [comment]
hmm, next is probably https - then better pngs.

**Comment by henrock** [comment]
I contributed with 0.01. All I could afford right now. :-\

**Comment by dominictarr** [comment]
update: new badge.

**Comment by dominictarr** [comment]
oh oops - I broke the github rate limit...

**Comment by dominictarr** [comment]
Okay, fixed github - but now blockchain.info is down.
Guess this is what happens when you depend on an external service.

**Comment by eins78** [comment]
> blockchain.info is down

side note: last time I looked, blockchain.info offered an API similar to that of a bitcoind instance running on an own server. by using that one could keep using the service for now, with an easy path for migration in the future.

**Comment by dominictarr** [comment]
Okay, blockchain.info is back up.

@eins78 good point - although I may need to upgrade my server to fit that much data.
last time I synced the blockchain it was nearly 20gb. Can you query any wallet with bitcoind?
Is there another option that doesn't sync the entire blockchain?

**Comment by ralphtheninja** [comment]
The bitcoind instance doesn't have to be on your server. So if you know someone you trust that's running a node, you could query that one.

**Comment by eins78** [comment]
@dominictarr For now, blockchain.info should be stable enough for this project. 
I was talking about this API: http://blockchain.info/api/json_rpc_api
It still talks to blockchain.info, but is using the same methods that would be used with an own instance (full node), so in case it is neccessary or wanted in the future just the API endpoint would need to be changed.
(Of course @ralphtheninja is also right, "own instance" does not mean "on the same server", just any full node than can be trusted and acessed)

**Comment by joates** [comment]
@dominictarr nice work on the badge, looks great.. i didn't realize yr **also** a _frontend ninja_

**Comment by joates** [comment]
i have some ideas about decoupling the api interface with a layer of abstraction and then promoting an ecosystem of userland plugins to provide the bindings to specific services (github / blockchain / anything else that develops in the future) but feeling a bit under the weather so will think on this and propose something in a new discussion when i'm feeling more lucid.

**Comment by ralphtheninja** [comment]
@joates That's a great idea. Perhaps a generic way of accessing the balance of coin X, so adding any other crypto currency would be trivial.

**Comment by dominictarr** [comment]
@joates you should post an issue under which to discuss that feature.

**Comment by dominictarr** [comment]
DONE.

**Comment by dominictarr** [comment]
reopening, because I realized there is a bug. feedopensource can't see closed issues.

**Comment by ralphtheninja** [comment]
Aha the D O N E is parsed by feedopensource to produce the checkmarks?

**Comment by dominictarr** [comment]
@ralphtheninja no, just whether the issue is closed.

**Comment by ralphtheninja** [comment]
@dominictarr Of course. Very stupid question from me, but sometimes I don't think before I ask :)

**Comment by NHQ** [comment]
PS http://blockexplorer.com/tx/b3df16687b4414f43926e2546de3edd90a4e6812f7110c8350da8941b91ec89b

**Comment by dominictarr** [comment]
@NHQ ah, cool thanks!
